CONTRIBUTING.md

```diff
@@ -97,33 +104,114 @@ Categories of system props are exported from `src/constants.js`:
 
+Component.propTypes = {
+   ...COMMON.propTypes
+}
+```
+
+### Linting
+
+We use the [React configuration](https://github.com/github/eslint-plugin-github/blob/master/lib/configs/react.js) from [GitHub's eslint plugin](https://github.com/github/eslint-plugin-github) to lint our JavaScript. To check your work before pushing, run:
+
+```
+yarn run lint
+```
+
+Or, you can use [npx] to run eslint on one or more specific files:
 
-## Deployment
-We deploy the Primer Components site using [Now]. Install the Now CLI and log in with:
 
 ```sh
-yarn global now
-now login
+# lint the component and the tests in src/__tests__
+npx eslint src/**/MyComponent.js
 ```
 
-Once you're logged in, sync your local git repo with the `master` branch and run:
+**Protip:** The [eslint `--fix` flag](https://eslint.org/docs/user-guide/command-line-interface#--fix) can automatically fix most linting errors, such as those involving whitespace or incorrect ordering of object keys and imports. You can fix those issues across the entire project with:
 
 ```sh
-script/deploy
+yarn run lint -- --fix
+```
+
+**Protip:** `yarn run lint -- --quiet` (or `npx eslint --quiet ...`) will suppress warnings so that you can focus on fixing errors.
+
+### Testing
+
+We test our components with [Jest](https://facebook.github.io/jest/) and [react-test-renderer](https://reactjs.org/docs/test-renderer.html). You can run the tests locally with `yarn test`. To run the tests as you work, run Jest in watch mode with:
+
+```
+yarn test -- --watch
 ```
 
-This will create a new deployment and alias it to its production URL, [primer-components.now.sh](https://primer-components.now.sh).
+See [`src/__tests__/example.js`](src/__tests__/example.js) for examples of ways that we test our components.
+
+### TypeScript support
+
+Several of the projects that consume Primer Components are written in TypeScript! Though Primer Components is not currently written in TS, we do export type definitions in order to make Primer Components compatible with other TS projects.
+
+Whenever adding new components or modifying the props of an existing component, **please make sure to update the type definition** in `index.d.ts`! This is super important to make sure we don't break compatibility :) 
+
+### Additional resources
+
+- [Primer Components Philosophy](https://primer.style/components/philosophy)
+- [Primer Components Core Concepts](https://primer.style/components/core-concepts)
+- [Primer Components System Props](https://primer.style/components/system-props)
+- [Styled Components docs](https://styled-components.com/)
+- [Styled System docs](https://styled-system.com/)
+
+## Writing documentation
+
+We use [Doctocat](https://github.com/primer/doctocat) to power our documentation site at [https://primer.style/components](https://primer.style/components/).
+
+To add a new component to our documentation site, create a new file with the `.md` extension for your component in `docs/content/components`.
```

Comment:
The path is actually `docs/content`. I think we did this to avoid having paths like `primer.style/components/components/Button`

```suggestion
To add a new component to our documentation site, create a new file with the `.md` extension for your component in `docs/content` (e.g. `docs/content/Button.md`).
```